README.md

```diff
@@ -61,3 +61,393 @@ java -jar target/sdk-1.0-SNAPSHOT-jar-with-dependencies.jar -authKey <keyPasswor
 ```
 java -jar target/sdk-1.0-SNAPSHOT-jar-with-dependencies.jar -cryptoKey <keyPassword>
 ```
+
+### Using the Citizen Java SDK
+
+The Java SDK allows a third-party entity to integrate their service with Citizen. It has functionality for handling tokens, configuration of authentication and encryption keys, and endpoints and login management. It can also be used to set up test accounts which can be used to validate configuration. Sample code of these operations is contained in Examples.java.
+
+### Building and installing the SDK
+
+The SDK can be built with the following command:
+
+```
+  mvn clean install
+```
+
+### Initialising the SDK
+
+The SDK is initialised by giving the connection details of the Citizen Service.
+
+```
+  CitizenApi citizenApi = new CitizenApi();
+  citizenApi.useDevelopmentApi();
+```
+
+A logging callback may also be registered. This returns a status code and log message. The code below registers an anonymous function:
+
+```
+  citizenApi.registerLoggingCallback((status, message) -> {
+      LOGGER.info("status: " + status + ", message: " + message);
+  });
+```
+
+### Generating Key Pairs
+
+The Citizen Service uses public/private key pairs for encryption and for authentication. RSA 2048 bit keys are used for encryption. Authentication uses ECDSA signatures with EC secp256r1 (aka P-256 aka prime256v1) keys. 
```

Comment:
Maybe mention what we call each in the SDK. So CryptoKeys for encryption/decryption and the AuthKeys for authentication. You've mentioned it below as well but we want to make it so that people don't somehow miss that bit of information and get confused.

Reply:
I added a mention about the two types of keys.